schematic/store/synapse.py

```diff
@@ -1467,24 +1585,89 @@ def upload_manifest_file(
                 + file_extension
             )
 
-        manifestSynapseFile = File(
-            metadataManifestPath,
-            description="Manifest for dataset " + datasetId,
-            parent=datasetId,
-            name=file_name_new,
+        manifest_synapse_file = None
+        try:
+            # Rename the file to file_name_new then revert
+            # This is to maintain the original file name in-case other code is
+            # expecting that the file exists with the original name
+            original_file_path = metadataManifestPath
+            new_file_path = os.path.join(
+                os.path.dirname(metadataManifestPath), file_name_new
+            )
+            os.rename(original_file_path, new_file_path)
+
+            manifest_synapse_file = self._store_file_for_manifest_upload(
+                new_file_path=new_file_path,
+                dataset_id=datasetId,
+                existing_file_name=file_name_full,
+                file_name_new=file_name_new,
+                restrict_manifest=restrict_manifest,
+            )
+            manifest_synapse_file_id = manifest_synapse_file.id
+
+        finally:
+            # Revert the file name back to the original
+            os.rename(new_file_path, original_file_path)
+
+            if manifest_synapse_file:
+                manifest_synapse_file.path = original_file_path
+
+        return manifest_synapse_file_id
+
+    def _store_file_for_manifest_upload(
+        self,
+        new_file_path: str,
+        dataset_id: str,
+        existing_file_name: str,
+        file_name_new: str,
+        restrict_manifest: bool,
+    ) -> File:
+        """Handles a create or update of a manifest file that is going to be uploaded.
+        If we already have a copy of the Entity in memory we will update that instance,
+        otherwise create a new File instance to be created in Synapse. Once stored
+        this will add the file to the `synapse_entity_tracker` for future reference.
+
+        Args:
+            new_file_path (str): The path to the new manifest file
+            dataset_id (str): The Synapse ID of the dataset the manifest is associated with
+            existing_file_name (str): The name of the existing file
+            file_name_new (str): The name of the new file
+            restrict_manifest (bool): Whether the manifest should be restricted
+
+        Returns:
+            File: The stored manifest file
+        """
+        local_tracked_file_instance = (
```

Comment:
I want to make sure that I understand the logic here correctly. Is the following correct? : 
1)  locate a file whose name matches existing_file_name
2) if the first step is None, then locate a file whose name matches file_name_new
3) if both are None, then local_tracked_file_instance is None

Reply:
Yes that's correct

Reply:
Thanks Bryan!